spec/api/repositories_spec.cr

```diff
@@ -0,0 +1,37 @@
+require "../helper"
+
+module PlaceOS::FrontendLoader::Api
+  describe Repositories do
+    describe "query" do
+      it "does not mutate the managed repositories" do
+        branch = "test-fixture"
+        checked_out_commit = "d37c34a49c96a2559408468b2b9458867cbf1329"
+        folder = UUID.random.to_s
+        repo = PlaceOS::Model::Generator.repository(type: :interface).tap do |r|
+          r.name = "compiler"
+          r.uri = "https://github.com/placeos/compiler"
+          r.username = "robot@place.tech"
+          r.branch = branch
+          r.commit_hash = checked_out_commit
+          r.folder_name = folder
+        end
```

Comment:
```suggestion
        repo = example_respoitory
```

Reply:
Couple of uses of this. If it's just the folder name that needs some variation, could it be added as an arg to that helper?

Reply:
Issue for PlaceOS/models? i’ll reduce duplication where i can here

Reply:
This ones local: https://github.com/PlaceOS/frontend-loader/blob/3520ec09aaedf25c621333c3fb77727b93297792/spec/helper.cr#L25